src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -380,21 +381,24 @@ export class MockHttpSocket extends MockSocket {
 
     if (response.body) {
       try {
-        const reader = response.body.getReader()
-
-        while (true) {
-          const { done, value } = await reader.read()
-
-          if (done) {
-            serverResponse.end()
-            break
-          }
-
-          serverResponse.write(value)
-        }
+        serverResponse.socket?.on('error', () => {})
+
+        await pipeline(
+          Readable.fromWeb(
+            response.body as webstream.ReadableStream<Uint8Array>
+          ),
+          serverResponse
+        )
       } catch (error) {
-        // Coerce response stream errors to 500 responses.
-        this.respondWith(createServerErrorResponse(error))
+        // We don't have direct access to the real IncomingMessage here to
+        // emit the 'error' event so we have to reach to the Socket internal state
+        // @ts-expect-error
+        this._httpMessage?.res?.emit('error', error)
```

Comment:
This is pretty annoying but we have to manually forward the error event to the actual `IncomingMessage` so that any `error` event handler the user has attached to it is called correctly.

Reply:
Let's store `IncomingMessage` in a variable and emit error on it directly maybe? So we don't have to claw our way to the same message reference here.